packages/blade/src/components/ListView/_decisions/decisions.md

```diff
@@ -0,0 +1,520 @@
+# List View Pattern API Decision
+
+> [!NOTE]
+>
+> Questions like what is pattern, why are we building these patterns, and scope of pattern are answered in [pattern-terminology.md](./pattern-terminology.md)
+
+List View Pattern is a UI pattern that includes Table, Filters, and Search of the table. It is a commonly used pattern on Razorpay dashboards to visualize data and help consumers filter / search through that data.
+
+![alt text](image-5.png)
+
+## API
+
+### With Layout Component
+
+```jsx
+<ListView>
+  <ListViewFilters
+    quickFilters={
+      <QuickFilterGroup onChange={({ values }) => console.log(values)}>
+        <QuickFilter title="Status: Completed" value="completed" />
+        <QuickFilter title="Status: Pending" value="pending" />
+      </QuickFilterGroup>
+    }
+    onSearchChange={() => {
+      /* filter on search */
+    }}
+  >
+    <FilterChipGroup>
+      <Dropdown>
+        <FilterChip />
+        <DropdownOverlay></DropdownOverlay>
+      </Dropdown>
+
+      <Menu>
+        <FilterChip />
+        <MenuOverlay>
+          <TextInput />
+        </MenuOverlay>
+      </Menu>
+    </FilterChipGroup>
+  </ListViewFilters>
+  <Table data={{ nodes: filteredNodes }} />
+</ListView>
```

Comment:
Hi @saurabhdaware , if possible please also mention how states such as loading, error states are handled in list views. Would it be handled out of the box by the ListView pattern, or would the consumers will need to implement it on their own.

Reply:
Oh good question. It will have to be handled on consumer-side since blade itself won't have context of your data fetching.

But I think we might have to define it on figma for designers. Let me check with our design team.